bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/api/TypeTags.java

```diff
@@ -33,7 +34,22 @@
     public static final int XML_TAG = JSON_TAG + 1;
     public static final int TABLE_TAG = XML_TAG + 1;
     public static final int NULL_TAG = TABLE_TAG + 1;
-    public static final int ANYDATA_TAG = NULL_TAG + 1;
+
+    // subtypes
+    public static final int SIGNED32_INT_TAG = NULL_TAG + 1;
+    public static final int SIGNED16_INT_TAG = SIGNED32_INT_TAG + 1;
+    public static final int SIGNED8_INT_TAG = SIGNED16_INT_TAG + 1;
+    public static final int UNSIGNED32_INT_TAG = SIGNED8_INT_TAG + 1;
+    public static final int UNSIGNED16_INT_TAG = UNSIGNED32_INT_TAG + 1;
+    public static final int UNSIGNED8_INT_TAG = UNSIGNED16_INT_TAG + 1;
+    public static final int CHAR_STRING_TAG = UNSIGNED8_INT_TAG + 1;
```

Comment:
lets add a comment break here
```suggestion
    // subtypes with no filler value
    public static final int CHAR_STRING_TAG = UNSIGNED8_INT_TAG + 1;
```

Reply:
looks like we can't say this, only no-filler-value is char

Reply:
Since the built-in subtypes of `xml` have filler values, we can't say that there. Anyway current implementation does not consider `built-in Xml subtypes` for filler values. It looks like a bug. 
e.g.
```
import ballerina/io;

public function main() {

    'xml:Text x1 = 'xml:createText("text 1");
    'xml:Text x2 = 'xml:createText("text 2");
    'xml:Text x3 = 'xml:createText("text 3");

    'xml:Text[] x = [x1, x2];
    insertListValue(x, 4, x3);
    io:println(x);
} 

function insertListValue('xml:Text[] list, int pos, 'xml:Text value) {
    list[pos] = value;
}
```
Above code panics abruptly with the following error message
`error: {ballerina/lang.array}IllegalListInsertion {"message":"array of length 2 cannot be expanded into array of length 5 without filler values"}`
But should allow the insertion with filler values according to the [spec](https://ballerina.io/spec/lang/master/#FillMember).

Reply:
Shall we create an issue for this if you haven't already and resolve it from this PR since I see that you have tried to fix it here